## Практическая работа №2 Теория вероятностей и математическая статистика

#### Задание №1
##### Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Для решения задачи воспользуемся формулой бернули
$$P(X=k) = C^k_{n}p^kq^{n-k}$$
$$p = 0.8; q = 1 - p = 0.2$$
$$P(X=85) = C^{85}_{100}p^{85}q^{15} = \frac{100!}{85!(100-85)!}*0.8^{85}*0.2^{15}$$
с такими числами проще написать python функцию для расчётов

In [22]:
import numpy as np
def bern(n:int, k:int, p:float) -> float:
    """
    Реализация формулы Бернули.
    :param n: Количество независящих испытаний
    :param k: Количество наступления события А
    :param p: Вероятность наступления события А
    :return: Вероятность наступления события А k раз в n испытаниях
    """
    return (np.math.factorial(n)/(np.math.factorial(k)*np.math.factorial(n-k)))*(p**k)*(1-p)**(n-k)

In [23]:
P = bern(100, 85, 0.8)
print('P(k=85) = %.4f' % P)

P(k=85) = 0.0481


---

**Ответ: Вероятность того что стрелок попадёт в мишень ровно 85 раз развна 0.0481 (4.81\%)**

---

#### Задание №2
##### Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Применим формулу Пуассона, зная, что m = 0 и $\lambda = np = 5000*0.0004 = 2$
$$P_{0} = \frac{\lambda^m}{m!}e^{-\lambda} = \frac{2^0}{0!}e^{-2} = 1*e^{-2} = 0.1353$$

##### Значит, вероятность того что перегорит одна лампочка в первый день равна 0.1353 (13.53%)

Найдём вероятность того что перегорят ровно 2 лампочки. m = 2
$$P_{2} = \frac{\lambda^m}{m!}e^{-\lambda} = \frac{2^2}{2!}e^{-2} = 2*e^{-2} = 0.2706 = 27.06\%$$
___

**Ответ:**
**Вероятность того что ни одна ламвочка не перегорит в первый день равна 0.1353 (13.53\%).**
**Вероятность того что перегорит ровно 2 лампочки равна 0.2706 (27.06\%)**

___

#### Задание №3
##### Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Воспользуемся формулой бернули, где, p = q = 0.5, n = 144, k =70
$$P(X=70) = C^{70}_{144}p^{70}q^{144-70} = \frac{144!}{70!(144-70)!}*0.5^{70}*0.5^{74}$$
Для расчётов воспользуемся ранее созданной функцией

In [27]:
print('P(k=70) = %.4f' % bern(144, 70, 0.5))

P(k=70) = 0.0628


---

**Ответ: Вероятность того что орел выпадет ровно 70 раз равна 0.0628 (6.28\%)**

---

#### Задание №4
##### В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.
Какова вероятность того, что все мячи белые?
Какова вероятность того, что ровно два мяча белые?
Какова вероятность того, что хотя бы один мяч белый?

**Какова вероятность того, что все мячи белые?**
Событие А - из первой корзины вытащили два белых мяча
Событие B - из второй корзины вытащили два белых мяча
$P(A+B) = P(A) \cdot P(B)$
Можно решить через условную вероятность, $P(A) = \frac{7}{10}\cdot\frac{6}{9}$, $P(B) = \frac{9}{11}\cdot\frac{8}{10}$
$$P(A+B) = \frac{7}{10}\cdot\frac{6}{9}\cdot\frac{9}{11}\cdot\frac{8}{10} = \frac{3024}{9900} = 0.3055 = 30.50\%$$

Или с помощью функций комбинаторики
$$P(A+B) = P(A) \cdot P(B) = \frac{C^2_7}{C^2_{10}}\cdot\frac{C^2_9}{C^2_{11}} = 0.3055$$

**Ответ: Вероятность того что все мячи белые 0.3055(30.55\%)**

---

**Какова вероятность того, что ровно два мяча белые?**
Для расчёта вероятности необходимо найти возможные комбинации решений и сложить их вероятности
1. Из первой корзины вытащили 2 белых ($A_1$), из второй 2 не белых ($A_2$)
2. Из первой корзины вытащили 1 белый и 1 не белый ($B_1$) и из второй корзины 1 белый и 1 не белый ($B_2$)
3. Из первой корзины вытащили 2 не белых ($C_1$), а из второй - 2 белых ($C_2$)

$$ P = P(A_1)\cdot P(A_2) + P(B_1) \cdot P(B_2) + P(C_1) \cdot P(C_2)$$
$$ P(A_1) \cdot P(A_2) = \frac{C^2_7}{C^2_{10}} \cdot \frac{C^0_9}{C^2_{11}} = \frac{\frac{7!}{2!(7-2)!}}{\frac{10!}{2!(10-2)!}} \cdot \frac{\frac{9!}{0!(9-0)!}}{\frac{11!}{2!(11-2)!}} = \frac{6*7}{7*10} \cdot \frac{1*2}{10*11} = \frac{42}{90}\cdot\frac{2}{110} =  0.4667 * 0.0182 = 0,0085$$

$$P(B_1)\cdot P(B_2) = \frac{C^1_7 \cdot C^1_3}{C^2_{10}} \cdot \frac{C^1_9\cdot C^1_2}{C^2_{11}} = \frac{\frac{7!}{1!(7-1)!}\cdot\frac{3!}{1!(3-1)!}}{\frac{10!}{2!(10-2)!}} \cdot \frac{\frac{9!}{1!(9-1)!}\cdot\frac{2!}{1!(2-1)!}}{\frac{11!}{2!(11-2)!}} = \frac{7*3}{45} \cdot \frac{9*2}{55} = 0.4667 * 0.3273 = 0,1528$$

$$P(C_1)\cdot P(C_2) = \frac{C^0_7}{C^2_{10}}\cdot\frac{C^2_9}{C^2_{11}} = \frac{\frac{7!}{0!(7-0)!}}{\frac{10!}{2!(10-2)!}} \cdot\frac{\frac{9!}{2!(9-2)!}}{\frac{11!}{2!(11-2)!}} = \frac{2}{9*10}\cdot\frac{4*9}{5*11} = 0.0222 * 0.6546 = 0.0145$$

$$P = 0.0085 + 0.1528 + 0.0145 = 0,1758 = 17.58\%$$

**Ответ: вероятность вытащить ровно два мяча равна 0.1758(17.58%)**

---

**Какова вероятность того, что хотя бы один мяч белый?**
Тут удобей пойти от противного и вычислить вероятность того, что все мячи которые вытащили не белые ($\bar A$)
$P(A) = 1 - P(\bar A)$

$$P(\bar A) = \frac{C^2_3}{C^2_{10}} \cdot \frac{C^2_2}{C^2_{11}} = \frac{\frac{3!}{2!(3-2)!}}{\frac{10!}{2!(10-2)!}} \cdot \frac{\frac{2!}{2!(2-2)!}}{\frac{11!}{2!(11-2)!}} = \frac{1}{15} \cdot \frac{1}{55} = 0.0667 * 0.0182 = 0,0012$$
$$P(A) = 1 - 0.0012 = 0,9988 = 99.88\%$$

**Ответ: вероятностьтого что хотя бы один мяч белый 0.9988 (99.88\%)**